src/highdicom/legacy/sop.py

```diff
@@ -1,20 +1,21 @@
-"""Module for SOP Classes of Legacy Converted Enhanced Image IODs."""
-
+""" Module for SOP Classes of Legacy Converted Enhanced Image IODs."""
 import logging
-from collections import defaultdict
-from typing import Any, Dict, List, Optional, Sequence
-
-from pydicom.datadict import tag_for_keyword
+from typing import Any, List, Union, Callable
+from numpy import log10, array, ceil, cross, dot, ndarray
+from pydicom.datadict import tag_for_keyword, dictionary_VR, keyword_for_tag
 from pydicom.dataset import Dataset
-
+from pydicom.tag import Tag, BaseTag
+from pydicom.dataelem import DataElement
+from pydicom.sequence import Sequence as DataElementSequence
+from pydicom.multival import MultiValue
+from datetime import datetime, timedelta
+from pydicom.valuerep import DT, DA, TM
+from copy import deepcopy
+from pydicom.uid import UID
 from highdicom.base import SOPClass
-from highdicom._iods import IOD_MODULE_MAP, SOP_CLASS_UID_IOD_KEY_MAP
+from highdicom._iods import IOD_MODULE_MAP
 from highdicom._modules import MODULE_ATTRIBUTE_MAP
-
-
-logger = logging.getLogger(__name__)
-
-
+# logger = logging.getLogger(__name__)
```

Comment:
There is a problem with declaring logger globally. Usually, the logger is configured by the end-user/progrmmer after importing the modules.  This scenario doesn't work with the global logger because the logger is instantiated before configuration. For instance in user code when they import highdicom (the global logger is instantiated) then configures the logger which would not affect the already instantiated global logger. I tested this myself and the logger failed to log for highdicom.

Reply:
Afshin, I do not understand your concern. The way I understand it, there is a separate logger for specific modules, in this case this will be a logger for `highdicom.sr.sop`. The user can access that same logger by name after importing the module. I don't think user application using highdicom would want to log into the same logger defined in the highdicom modules, but the user could, if needed.

Reply:
To my understanding, the whole point of using loggers is giving the control of how to log (configuration) to the end-user. Otherwise printing on std output would suffice. The configuration says root logger how(style) and where(file/std/network) to log which is not done by the app developer but by the end-user. Usually, I configure the root logger once and get the logger in each function with the module name. If a specific configuration for the module name exists it is configured in that way otherwise it will be configured as root. 
With highgdicom module logger, when I want to configure the logger, I need to access the instantiated global logger separately and configure it all over again which imposes extra coding and a dirtied control flow. Instead, each function could get the once configured logger while it is being called. I'm also curious what is the benefit of having a global logger in one module over getting a logger with the module name separately in function while it complies by the most recent setting?

Reply:
Let's hear from @hackermd, but we cannot revert the decisions made already in the library without first consulting with Markus. If that's how he would like logging to be done, then that's how it should be done. I suggest that changes to logging, if needed, should be revisited later in a separate PR.

Reply:
It was just a suggestion. It started when I was not able to get logging from highdicom and change the code, I thought it wouldn't change the coding structure too much. Any way if @hackermd insists on keeping this way, I'll comply.

Reply:
The convention in Python is to instantiate a logger instance per module in the library. Logging is then configured only once at the main entry point of the application (e.g., command line program). See [Python logging tutorial](https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial).

Reply:
Well, there is just one logger for each module even if I instantiate the logger within the function because I use the module name even in the function. Anyway, when I configure the logger and highdicom's logger is global, I don't get any log from highdicom at all.

Reply:
How do you configure logging?

I generally do something along those lines:

```python
import logging
import sys

if __name__ == '__main__':
    logger = logging.getLogger('highdicom')
    stderr_handler = logging.StreamHandler(stream=sys.stderr)
    logger.addHandler(stderr_handler)
    logger.setLevel(logging.DEBUG)
```

See also [configuring logging for a library](https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library).

Note that we use logging sparsely in *highdicom*, so you will generally not get a lot of log messages.

Reply:
I configure logging by a .json file. I configure each module separately, I just configure the root and every module will instantiate the root logger. I also re-implemented the logging handler to work on multiprocessing fashion. This policy works for my code and a couple of google cloud api but doesn't work for highdicom unless I instantiate logger in the function. No worries, I will find a workaround for this problem of mine and apply your suggestion.